habitat-lab/habitat/datasets/rearrange/samplers/object_sampler.py

```diff
@@ -237,13 +237,13 @@ def sample_placement(
         """
         num_placement_tries = 0
         new_object = None
-        navmesh_vertices = np.stack(
-            sim.pathfinder.build_navmesh_vertices(), axis=0
-        )
-        # Note: we cache the largest island to reject samples which are primarily accessible from disconnected navmesh regions. This assumption limits sampling to the largest navigable component of any scene.
-        self.largest_island_size = max(
-            [sim.pathfinder.island_radius(p) for p in navmesh_vertices]
+
+        # Note: we cache the largest island ID to reject samples which are primarily accessible from disconnected navmesh regions.
+        # This assumption limits sampling to the largest navigable component of any scene.
+        island_areas = list(
+            map(sim.pathfinder.island_area, range(sim.pathfinder.num_islands))
         )
+        self.largest_island_id = island_areas.index(max(island_areas))
```

Comment:
Seems wasteful to compute this every time we sample an object. Instead, we should check if `self.largest_island_id` is unset and compute it once, then cache the result until `reset` is called.